### PR TITLE
Storage API Changes (Part 2 - Simplify the Storage user experience)

### PR DESCRIPTION
**Pull request by tg-msft**
I'm adding this on top of the existing Storage API changes, so please only look at the last commit (see #6718 for Part 1):

- Rename `BlobClient` => `Specialized.BlobBaseClient`
- Add `BlobClient : BlobBaseClient` with an `Upload` method for block blobs
- Add overloads to ease parameter choice for `BlobBaseClient.Download`

### CODE REVIEW

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs [hunk 1]
```diff
@@ -0,0 +1,2191 @@
...
+
+        /// <summary>
+        /// Creates a clone of this instance that references a version ID rather than the base blob.
+        /// </summary>
+        /// /// <remarks>
+        /// Pass null or empty string to remove the verion ID returning a URL to the base blob.
+        /// </remarks>
+        /// <param name="versionId">The version ID to use on this blob. An empty string or null indicates to use the base blob.</param>
+        /// <returns>The new <see cref="BlobClient"/> instance referencing the verionId.</returns>
+        //public virtual BlobBaseClient WithVersionId(string versionId) => this.WithVersionIdImpl(versionId);
```
Comment (kfarmer-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs]: Status of the commented methods?

**Review by kfarmer-msft — APPROVED** [review]
Question about commented methods.

File: sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs [hunk 2]
```diff
@@ -49,1522 +68,334 @@ public class BlobClient
...
+            IProgress<StorageProgress> progressHandler = default,
+            CancellationToken cancellationToken = default) =>
+            this.UploadAsync(
+                content,
+                blobAccessConditions: blobAccessConditions,
+                blobHttpHeaders: blobHttpHeaders,
+                metadata: metadata,
+                progressHandler: progressHandler,
+                cancellationToken: cancellationToken)
+                .EnsureCompleted();
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs]: This calls async method without passing isAsync

File: sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs (hunk 1 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs]: These were commented out by Sean right before //build.

File: sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs (hunk 2 above)
Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs]: Good catch - fixed.

**Review by KrzysztofCwalina — APPROVED** [review]

File: sdk/storage/Azure.Storage.Blobs/tests/SessionRecords/ServiceClientTests/SetPropertiesAsyncAsync.json [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        ],
+        "x-ms-client-request-id": "56350fd505ab7c1c6d2bde63cdd902d3",
+        "x-ms-date": "Mon, 01 Jul 2019 21:58:04 GMT",
+        "x-ms-return-client-request-id": "true",
+        "x-ms-version": "2018-11-09"
+      },
+      "RequestBody": [
+        "\u003cStorageServiceProperties\u003e\r\n",
+        "  \u003cLogging\u003e\r\n",
+        "    \u003cVersion\u003e1.0\u003c/Version\u003e\r\n",
```
Comment (pakrym) [inline sdk/storage/Azure.Storage.Blobs/tests/SessionRecords/ServiceClientTests/SetPropertiesAsyncAsync.json]: I've noticed that the request body is pretty-printed, is it worth disabling the pretty-printing to save bandwidth?

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/SessionRecords/ServiceClientTests/SetPropertiesAsyncAsync.json]: These are data for recorded tests, correct?  If so, I don't think we are actually using any bandwidth.

Comment (pakrym) [reply sdk/storage/Azure.Storage.Blobs/tests/SessionRecords/ServiceClientTests/SetPropertiesAsyncAsync.json]: Recorded tests store exact thing that was sent over the wire during the Live test.

Comment (tg-msft) [reply sdk/storage/Azure.Storage.Blobs/tests/SessionRecords/ServiceClientTests/SetPropertiesAsyncAsync.json]: Yes - filed #6806.

**Pull request merged**